junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/Node.java

```diff
@@ -159,6 +159,11 @@ public boolean isSkipped() {
 		public Optional<String> getReason() {
 			return Optional.ofNullable(this.reason);
 		}
+
+		@Override
+		public String toString() {
+			return skipped ? String.format("Skip [%s]", reason) : "Do not skip";
+		}
```

Comment:
Will do. Quick question on JUnit's stance on `Optional` fields - do you still do that? `ConditionEvaluationResult` uses one (`reason`) and `SkipResult` might for the same, err, `reason`. Personally I like that style - purposely ignoring the JDK team's recommendations - but I wanted to make sure that this is not just an old artifact.